server/migrator/Importer.java

```diff
@@ -0,0 +1,311 @@
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package grakn.core.server.migrator;
+
+import com.google.protobuf.Parser;
+import grakn.common.collection.Pair;
+import grakn.core.Grakn;
+import grakn.core.common.exception.GraknException;
+import grakn.core.common.parameters.Arguments;
+import grakn.core.concept.thing.Attribute;
+import grakn.core.concept.thing.Entity;
+import grakn.core.concept.thing.Relation;
+import grakn.core.concept.thing.Thing;
+import grakn.core.concept.type.AttributeType;
+import grakn.core.concept.type.EntityType;
+import grakn.core.concept.type.RelationType;
+import grakn.core.concept.type.RoleType;
+import grakn.core.server.Version;
+import grakn.core.server.migrator.proto.DataProto;
+import grakn.core.server.migrator.proto.MigratorProto;
+import grakn.core.server.rpc.MigratorRPCService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static grakn.core.common.exception.ErrorMessage.Migrator.FILE_NOT_READABLE;
+import static grakn.core.common.exception.ErrorMessage.Migrator.INVALID_DATA;
+import static grakn.core.common.exception.ErrorMessage.Migrator.TYPE_NOT_FOUND;
+
+public class Importer {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MigratorRPCService.class);
+    private static final Parser<DataProto.Item> ITEM_PARSER = DataProto.Item.parser();
+    private static final int BATCH_SIZE = 10_000;
+    private final Grakn.Session session;
+    private final Path filename;
+    private final Map<String, String> remapLabels;
+
+    private final Map<String, byte[]> idMap = new HashMap<>();
+    private final List<Pair<byte[], List<String>>> missingOwnerships = new ArrayList<>();
+    private final List<Pair<byte[], List<Pair<String, List<String>>>>> missingRolePlayers = new ArrayList<>();
+    private long totalThingCount = 0;
+    private long entityCount = 0;
+    private long relationCount = 0;
+    private long attributeCount = 0;
+    private long ownershipCount = 0;
+    private long playerCount = 0;
+    private Grakn.Transaction tx;
+    private int commitWriteCount = 0;
```

Comment:
We removed `caches` compared with https://github.com/graknlabs/grakn/blob/1.8.4/server/migrate/Import.java, since hypergraph is already doing the cache.